storage/rocksdb/ha_rocksdb.cc

```diff
@@ -15494,6 +15517,10 @@ int ha_rocksdb::mrr_fill_buffer() {
   // The buffer has space for this many elements:
   ssize_t n_elements = (mrr_buf.buffer_end - mrr_buf.buffer) / element_size;
 
+  THD *thd = table->in_use;
+  ssize_t elements_limit = THDVAR(thd, mrr_batch_size);
+  n_elements = std::min(n_elements, elements_limit);
+
```

Comment:
Ok, now the memory overrun for @@rocksdb_mrr_batch_size=0 is gone but it will hit the assertion failure a few lines below.

I think this also needs to be fixed. Could you please also add the check for
`@@rocksdb_mrr_batch_size!=0` into this check into ha_rocksdb::multi_range_read_init
```
  if (!current_thd->optimizer_switch_flag(OPTIMIZER_SWITCH_MRR) ||
      (mode & HA_MRR_USE_DEFAULT_IMPL) ||
      (buf->buffer_end - buf->buffer < mrr_get_length_per_rec()) ||
      (active_index != table->s->primary_key && (mode & HA_MRR_SORTED))) {
```
The check already has the logic "if nothing fits into the buffer, switch to  the default implementation" (on line#3). I think it makes sense to apply it for rocksdb_mrr_batch_size=0, too.

Reply:
Good point! I added it.